numpy/core/arrayprint.py

```diff
@@ -606,6 +597,10 @@ def _formatArray(a, format_function, rank, max_line_len,
     return s
 
 class FloatFormat(object):
+
+    _nan_str = _options['_nan_str']
```

Comment:
Hmm, is this set when the module is executed on load, so this is a behavior change as the option is no longer dynamically set when the class is instantiated.